bin/citrea/tests/evm/web3.py_tests/test.py

```diff
@@ -0,0 +1,242 @@
+        contract = self.web3.eth.contract(abi=abi, bytecode=initcode)
+        tx = contract.constructor().build_transaction({
+            'from': "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266",
+            'value': 0,
+            'gas': 2000000,
+            'gasPrice': self.web3.to_wei(1, 'gwei'),
+            'nonce': self.web3.eth.get_transaction_count("0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266"),
+            'chainId': 5655
+        })
+        signed_tx = self.web3.eth.account.sign_transaction(tx, "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80")
+        tx_hash = self.web3.eth.send_raw_transaction(signed_tx.rawTransaction)
+        receipt = self.web3.eth.wait_for_transaction_receipt(tx_hash)
+        weth_address = receipt['contractAddress']
+
+        contract = self.web3.eth.contract(address=weth_address, abi=abi)
+        self.assertEqual(contract.functions.name().call(), "Wrapped Ether")
+        self.assertEqual(contract.functions.balanceOf("0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266").call(), 0)
+
+        # Deposit 1 Ether
+        tx = contract.functions.deposit().build_transaction({
+            'from': "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266",
+            'value': self.web3.to_wei(1, 'ether'),
+            'gas': 2000000,
+            'gasPrice': self.web3.to_wei(1, 'gwei'),
+            'nonce': self.web3.eth.get_transaction_count("0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266"),
+            'chainId': 5655
+        })
+        signed_tx = self.web3.eth.account.sign_transaction(tx, "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80")
+        tx_hash = self.web3.eth.send_raw_transaction(signed_tx.rawTransaction)
+        receipt = self.web3.eth.wait_for_transaction_receipt(tx_hash)
+
+        # Check if 1 WETH is minted
+        self.assertEqual(contract.functions.balanceOf("0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266").call(), self.web3.to_wei(1, 'ether'))
+
+    def test_call_errors_correctly(self):
+        try:
+            selector = self.web3.keccak(text='ERRONEUS_FUNC()')[:4]
+            self.web3.eth.call({'value': 0, 'to': '0x3100000000000000000000000000000000000001', 'data': selector})
+            self.fail("Expected call to fail, but it succeeded.")
+        except Exception as e:
+            self.assertEqual(str(e), "('execution reverted', 'no data')")
```

Comment:
should we be comparing e.to_string()?

is there another error type?

we should be testing if our error can be converted to the expected type in the library